Mixin/UserInterface/Controllers/Common/ScreenLock/ScreenLockManager.swift

```diff
@@ -0,0 +1,175 @@
+import UIKit
+import SnapKit
+import LocalAuthentication
+import MixinServices
+
+final class ScreenLockManager {
+    
+    private enum State: String {
+        case none
+        case authenticationFailed
+        case authenticationSucceed
+        case didEnterBackground
+        case willResignActive
+        case didBecomeActive
+    }
+    
+    static let shared = ScreenLockManager()
+    
+    private(set) var validAfterLastAuthentication = false
+    private(set) var window: UIWindow?
+    
+    private var viewController: ScreenLockViewController?
+    private var hasLastBiometricAuthenticationFailed = false
```

Comment:
Can these two variables `validAfterLastAuthentication` and `hasLastBiometricAuthenticationFailed` be merged into one?